speculation-rules/prerender/resources/utils.js

```diff
@@ -143,3 +143,64 @@ function createFrame(url) {
       document.body.appendChild(frame);
     });
 }
+
+class PrerenderChannel extends EventTarget {
+  broadcastChannel = null;
+
+  constructor(uid, name) {
+    super();
+    this.broadcastChannel = new BroadcastChannel(`${uid}-${name}`);
+    this.broadcastChannel.addEventListener('message', e => {
+      this.dispatchEvent(new CustomEvent('message', {detail: e.data}));
+    });
+  }
+
+  postMessage(message) {
+    this.broadcastChannel.postMessage(message);
+  }
+
+  close() {
+    this.broadcastChannel.close();
+  }
+};
+
+async function create_prerendered_page(t) {
+  const uuid = token();
+  new PrerenderChannel(uuid, 'log').addEventListener('message', message => {
+    if(window.console)
+      console['log']('[From Prerendered]', ...message.detail);
```

Comment:
Would you add a comment why we need to use `console['log']` instead of `console.log`?